http/src/main/scala/org/http4s/blaze/http/http20/ByteBufferInputStream.scala

```diff
@@ -0,0 +1,42 @@
+package org.http4s.blaze.http.http20
+
+import java.io.{IOException, InputStream}
+import java.nio.ByteBuffer
+
+final class ByteBufferInputStream(buffer: ByteBuffer) extends InputStream {
+
+  private var markSize = 0
+
+  override def read(): Int = {
+    markSize -= 1
+    buffer.get()
```

Comment:
Yes, your right. Fortunately (maybe unfortunately) the twitter header decoder, which requires an ```InputStream```, always checks if data is available before reading, and converts the result back to a byte, so it avoids the [0,256) concern. Better to fix it in case they change their impl, and good catch.
This class will also be made private. There are numerous better solutions out there if someone needs one; I didn't want to pull in a dep for just this.

One such trick to get the unsigned range is ```buffer.get() & 0xff```